autoscale/src/main/java/com/sequenceiq/periscope/utils/StackResponseUtils.java

```diff
@@ -33,9 +33,7 @@
                 .filter(instanceGroupV4Response -> instanceGroupV4Response.getName().equalsIgnoreCase(hostGroup))
                 .flatMap(instanceGroupV4Response -> instanceGroupV4Response.getMetadata().stream())
                 .filter(instanceMetaData -> instanceMetaData.getDiscoveryFQDN() != null)
-                // TODO CB-14929: the computation of host group counts depends on the scaling mechanism being used.
-                //  Even for STOPPED instances, this may require additional checks on status of the nodes.
-                .filter(instanceMetaData -> instanceMetaData.getInstanceStatus() != InstanceStatus.STOPPED)
+                .filter(instanceMetaData -> InstanceStatus.SERVICES_HEALTHY.equals(instanceMetaData.getInstanceStatus()))
```

Comment:
This will need additional changes. Not sure if planned in this PR or a separate PR.

Ref: the comment in YarnLoadEvaluator.

The same method is used to find hosts when triggering downscale and upscale, and at this point also for stop-start vs regular scaling. Both need to be thought through independently.

Would be good to map this out given potential states. From the YarnLoadEvaluator comment (assuming SERVICES_RUNNING instead of SERVICES_STARTING) - this leads to unnecessary scale operations.

Reply:
Yes, as discussed, will address the different possibilities in StackResponseUtils and YarnLoadEvaluator as part of a separate PR